File: static/java/FormulaAttachmentsField.java
// ==========================================
// MyAirtable static runtime.
// ==========================================

package myairtable;

/** Formula builder for attachment fields. */
public final class FormulaAttachmentsField implements FormulaField {

  private final String fieldId;

  public FormulaAttachmentsField(String fieldId) {
    this.fieldId = fieldId;
  }

  @Override
  public String getFieldId() {
    return fieldId;
  }

  /** Has exactly N attachments. */
  public String count(int n) {
    return "LEN(" + field() + ")=" + n;
  }

  /** Has no attachments. */
  @Override
  public String empty() {
    return "LEN(" + field() + ")=0";
  }

  /** Has attachments. */
  @Override
  public String notEmpty() {
    return "LEN(" + field() + ")>0";
  }
}

File: static/csharp/FormulaAttachmentsField.cs
namespace MyAirtable;

/// <summary>Formula builder for attachment fields.</summary>
public sealed class FormulaAttachmentsField : FormulaField
{
    public FormulaAttachmentsField(string fieldId)
        : base(fieldId) { }

    /// <summary>Has exactly N attachments.</summary>
    public string Count(int n) => "LEN(" + Field() + ")=" + n;

    /// <summary>Has no attachments.</summary>
    public override string Empty() => "LEN(" + Field() + ")=0";

    /// <summary>Has attachments.</summary>
    public override string NotEmpty() => "LEN(" + Field() + ")>0";
}
